lib/views.js

```diff
@@ -377,43 +376,39 @@ exports.tasks_messages = {
   }
 };
 
-// TODO refactor this, it can be much cleaner
 exports.tasks_pending = {
   map: function (doc) {
-    var has_pending,
-      tasks = doc.tasks || [],
-      scheduled_tasks = doc.scheduled_tasks || [];
-
     /*
      * Required fields for message to be processed:
      *  task `state` value must be 'pending'
      *  message needs the `to` and `message` properties
      */
     function hasPending(tasks) {
-      var has = false,
-        tasks = tasks || [];
+      var tasks = tasks || [];
+
       tasks.forEach(function(task) {
         if (task && task.state === 'pending') {
           task.messages.forEach(function(msg) {
             if (msg && msg.to && msg.message) {
-              has = true;
+              return true;
```

Comment:
I think this is incorrect. This is returning from the loop function, not returning from the view function. Either change the `forEach` to a `for(var i; ...)` or put it back the way it was.